src/test/java/com/google/api/codegen/config/ServiceConfigTest.java

```diff
@@ -0,0 +1,87 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.config;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.Authentication;
+import com.google.api.AuthenticationRule;
+import com.google.api.OAuthRequirements;
+import com.google.api.Service;
+import com.google.api.codegen.util.ProtoParser;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Model;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class ServiceConfigTest {
+  private static final Model protoModel = Mockito.mock(Model.class);
+  private static final ProtoApiModel apiModel = new ProtoApiModel(protoModel);
+  private static final ProtoParser protoParser = Mockito.mock(ProtoParser.class);
+  private static final Interface libraryService = Mockito.mock(Interface.class);
+  private static final Interface foodService = Mockito.mock(Interface.class);
+
+  private static final Service serviceYaml =
+      Service.newBuilder()
+          .setAuthentication(
+              Authentication.newBuilder()
+                  .addRules(
+                      AuthenticationRule.newBuilder()
+                          .setOauth(
+                              OAuthRequirements.newBuilder()
+                                  .setCanonicalScopes(
+                                      "https://www.googleapis.com/auth/calendar,\n"
+                                          + "https://www.googleapis.com/auth/calendar.read")))
+                  .addRules(
+                      AuthenticationRule.newBuilder()
+                          .setOauth(
+                              OAuthRequirements.newBuilder()
+                                  .setCanonicalScopes(
+                                      "https://www.googleapis.com/auth/default, "
+                                          + "https://www.googleapis.com/auth/pumpkin"))))
+          .build();
+
+  @BeforeClass
+  public static void startUp() {
+    Mockito.when(protoParser.getAuthScopes(libraryService))
+        .thenReturn(
+            Arrays.asList(
+                "https://www.googleapis.com/auth/library",
+                "https://www.googleapis.com/auth/default"));
+    Mockito.when(protoParser.getAuthScopes(foodService))
+        .thenReturn(
+            Arrays.asList(
+                "https://www.googleapis.com/auth/food", "https://www.googleapis.com/auth/default"));
+    Mockito.when(protoModel.getServiceConfig()).thenReturn(serviceYaml);
+  }
+
+  @Test
+  public void testServiceConfigFromProtoFile() {
+    List<ProtoInterfaceModel> interfaces =
+        Arrays.asList(
+            new ProtoInterfaceModel(libraryService), new ProtoInterfaceModel(foodService));
+    List<String> scopes = apiModel.getAuthScopes(protoParser, interfaces);
```

Comment:
Note, `apiModel` is a mock object, which depends only on other mock objects (except `ProtoInterfaceModel`, but that class was not changed in this PR, and it cannot be the subject of testing here). Thus the following assert statements check that mocks return proper values, but do not check that the actual thing.

Reply:
Correct, this test tests `ProtoApiModel#getAuthScopes()` on the surface, and only verifies that the resulting `List<String>` contains the Strings that we expect, given whatever the mocks return.

I will soon have out a PR for baseline tests for testing the ProtoParser functionality.

Reply:
`apiModel` is not a mock object because it was created using the `ProtoApiModel` constructor.

ProtoParserTest has been created to test ProtoParser.

Reply:
My apologies for misreading that apiModel is created using `ProtoApiModel` constructor. I "double checked" multiple times before making such a strong statement, but somehow I "saw" all being mocks in the initialization block. (maybe because ProtoApiModel has Model in its name and I read it  as "Mock" (looks similar)). With ProtoApiModel being initialized properly this test makes much more sense, but then it tests specifically `ProtoApiModel` class and `getAuthScopes`. Why isn't this test called `ProtoApiModelTest`, and the test method is not called `testGetAuthScopes()`?